### PR TITLE
Integrate CDC into k8ssandra-operator reconciliation logic and APIs

### LINKED ISSUE #570: K8SSAND-1568 ⁃ Add CDC integration to k8ssandra-operator
Once we have CDC integrated into cass-operator, we should ensure that the relevant API fields are exposed via k8ssandra-operator also.

(I was hoping to have this under epic [K8SSAND-1547](https://app.zenhub.com/workspace/o/k8ssandra/cass-operator/issues/343) but it lives in cass-operator, and you can't span an epic over multiple repos.)

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1568) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1568
┆priority: Medium

### PR DESCRIPTION
**Pull request by Miles-Garnsey**
**What this PR does**:

What it says on the tin.

Note that I've modified go.mod to use my recent changes to cass-operator. We need to revert this back to using a non-snapshot release prior to merge, but this is required because the cass-operator changes aren't yet merged.

**Which issue(s) this PR fixes**:
Fixes #570 

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/k8ssandra/v1alpha1/k8ssandracluster_types.go
Comment (adejanovski) [inline apis/k8ssandra/v1alpha1/k8ssandracluster_types.go:313]:
```suggestion
	// from where they can be exported to external systems.
```

File: pkg/cassandra/datacenter.go
Comment (adejanovski) [inline pkg/cassandra/datacenter.go:338]:
```suggestion
	if dcTemplate.CDC == nil {
		dcConfig.CDC = clusterTemplate.CDC
	} else {
	        dcConfig.CDC = dcTemplate.CDC
	}
```

**Review by adejanovski — CHANGES_REQUESTED** [review]
The coalesce logic between cluster and dc templates doesn't seem to be applied as it should. Let me know if I'm missing something.

Otherwise, LGTM

File: pkg/cassandra/datacenter.go
Comment (Miles-Garnsey) [reply pkg/cassandra/datacenter.go:338]: No, no. I intended exactly what I'd written. We don't allow definition of CDC feeds cluster-wide, they are defined per DC. Think about where you want the data to flow (and how many copies you want to land in each Pulsar cluster) and you'll understand why.

**Review by adejanovski — APPROVED** [review]
All good then 👍

**Comment by Miles-Garnsey** [comment]
**NOTE: DO NOT MERGE THIS PR** until changes are merged into cass-operator and we can reference a stable cass-operator version in `go.mod`.

**Comment by Miles-Garnsey** [comment]
I have bumped the cass-operator version in go.mod to point at the current master. I have also rebased.

Pending the tests passing (it was a gnarly rebase) I think this is ready for merge. @adejanovski do you agree?

**Comment by adejanovski** [comment]
> I have bumped the cass-operator version in go.mod to point at the current master. I have also rebased.
> 
> Pending the tests passing (it was a gnarly rebase) I think this is ready for merge. @adejanovski do you agree?

Could you instead rebase on top of main? The cass-operator 1.12.0 upgrade was just merged there.

**Comment by Miles-Garnsey** [comment]
> Could you instead rebase on top of main? The cass-operator 1.12.0 upgrade was just merged there.

@adejanovski that's what I'm saying, this is rebased on main. I'm now seeing the cert-manager issues you've been reporting though since that rebase.

**Comment by adejanovski** [comment]
> > Could you instead rebase on top of main? The cass-operator 1.12.0 upgrade was just merged there.
> 
> @adejanovski that's what I'm saying, this is rebased on main. I'm now seeing the cert-manager issues you've been reporting though since that rebase.

Nope, you're using the latest `main` [commit](https://github.com/k8ssandra/k8ssandra-operator/pull/580/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R15), when we want `v1.12.0` specifically until we cut out the release. While you can argue it's the same commit, it feels cleaner that the 1.2 release references cass-operator v1.12.0 specifically.

**Comment by Miles-Garnsey** [comment]
> Nope, you're using the latest main [commit](https://github.com/k8ssandra/k8ssandra-operator/pull/580/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R15), when we want v1.12.0 specifically until we cut out the release. While you can argue it's the same commit, it feels cleaner that the 1.2 release references cass-operator v1.12.0 specifically.

I agree that the release should use a real version instead of some commit on master. Since 1.12 is now released I've updated this PR again to point at it, and rebased again to try to resolver these cert-manager issues we've suddenly started seeing.

If the tests pass now this PR should be good to go.

**Comment by adejanovski** [comment]
@Miles-Garnsey, we're still good for a merge and CI is all green (yay).
I'll merge this later today if you don't get a chance to.

**Pull request merged**
